### ISSUE TITLE
Including item_features predicts same score for every item

### ISSUE THREAD

**Issue by pratik-a-sudo** [lyst/lightfm#581]
@maciejkula Hi Team,

I am currently exploring reco model which can produced better results as compared to ALS algo that we currently have in production. Lightfm allows us to add item and user features along with sample weights which provides advantage over ALS. Results of lightfm without using features i.e pure CF approach would yield similar results as ALS, hence we decided to item feature, but while using item feature (Genre in our case) in the model i seem to get same score for every item for each user, hence accuracy is too low and poor quality of recommendations. Below is code i am using

##########  Model

model = LightFM(loss='warp',
                item_alpha = 0.0000001,
                user_alpha = 0,
                    random_state=2016)

model = model.fit(user_show_interaction_csr,
                  sample_weight = weight_matrix_csr,
                  item_features=shows_metadata_csr,
                  #user_features=user_features_csr,
                  epochs=100)

#model.item_biases = np.zeros_like(model.item_biases)
#model.user_biases = np.zeros_like(model.user_biases)

#model.item_biases = 0.1*(model.item_biases)
#model.item_biases = model.item_biases.clip(0.01, 10)

item_biases, item_factors = model.get_item_representations(features=shows_metadata_csr)
user_biases, user_factors = model.get_user_representations()

# combine item_factors with biases for dot product
item_factors = np.concatenate((item_factors, np.ones((item_biases.shape[0], 1))), axis=1)
item_factors = np.concatenate((item_factors, item_biases.reshape(-1, 1)), axis=1)

# combine user_factors with biases for dot product  
user_factors = np.concatenate((user_factors, user_biases.reshape(-1, 1)), axis=1)
user_factors = np.concatenate((user_factors, np.ones((user_biases.shape[0], 1))), axis=1)
        
scores = user_factors.dot(item_factors.T)
######################################################

Our aim is to compare top 5 reco from the two models and validate it based on what the user actually watched.
scores[user_0] gives constant  value for all top 5 reco items, good thing is that it predicts dominant genre but since all item are equal recommended, how can we sort the item recommendations.

Does the above interpretation correct, or any changes that i can apply for this problem

**Comment by pratik-a-sudo** [comment]
Sorry, attaching identity matrix with features solved this issue

**Comment by abhisekpanda0** [comment]
@pratik-a-sudo Could you please explain in more detail how did you attach identity matrix with features?
